crypto/conf/conf_mod.c

```diff
@@ -161,16 +161,21 @@ int CONF_modules_load(const CONF *cnf, const char *appname,
 
     for (i = 0; i < sk_CONF_VALUE_num(values); i++) {
         vl = sk_CONF_VALUE_value(values, i);
-        ERR_set_mark();
+        /*
+         * we have to condition our mark setting here
+         * because parsing failures in the provider config
+         * aren't considered fatal and return success
+         */
+        if (flags & CONF_MFLAGS_IGNORE_ERRORS)
+            ERR_set_mark();
         ret = module_run(cnf, vl->name, vl->value, flags);
         OSSL_TRACE3(CONF, "Running module %s (%s) returned %d\n",
                     vl->name, vl->value, ret);
+
         if (ret <= 0)
-            if (!(flags & CONF_MFLAGS_IGNORE_ERRORS)) {
-                ERR_clear_last_mark();
-                return ret;
-            }
-        ERR_pop_to_mark();
+            return ret;
+        if (flags & CONF_MFLAGS_IGNORE_ERRORS)
+            ERR_pop_to_mark();
```

Comment:
You cannot leave the mark on the stack so if you want to keep the error on the stack you must call ERR_clear_last_mark();
```suggestion
            if (flags & CONF_MFLAGS_IGNORE_ERRORS)
                ERR_clear_last_mark();
            return ret;
        if (flags & CONF_MFLAGS_IGNORE_ERRORS)
            ERR_pop_to_mark();
```